crates/gosub_html5/src/parser/tree_builder.rs

```diff
@@ -25,3 +25,101 @@ pub trait TreeBuilder {
     /// Insert/update an attribute for an element node.
     fn insert_attribute(&mut self, key: &str, value: &str, element_id: NodeId) -> Result<()>;
 }
+
+#[cfg(test)]
+mod tests {
+    use gosub_testing::testing::tree_construction::fixture::{
+        fixture_root_path, read_fixture_from_path,
+    };
+    use gosub_testing::testing::tree_construction::Harness;
+    use test_case::test_case;
```

Comment:
It seems to not like these imports for some reason. Where is the `test_case` macro being defined?

Reply:
It's an external crate. Basically just run

`cargo add test_case --dev -p gosub_html5`

It also seems like fmt and clippy aren't happy, it can be fixed via `make format`

I haven't looked really into what is wrong, again I'll do that tomorrow.